src/main/java/dev/_2lstudios/chatsentinel/velocity/listeners/ChatListener.java

```diff
@@ -0,0 +1,165 @@
+package dev._2lstudios.chatsentinel.velocity.listeners;
+
+import com.velocitypowered.api.command.CommandSource;
+import com.velocitypowered.api.event.Subscribe;
+import com.velocitypowered.api.event.player.PlayerChatEvent;
+import com.velocitypowered.api.proxy.ConsoleCommandSource;
+import com.velocitypowered.api.proxy.Player;
+import com.velocitypowered.api.proxy.ProxyServer;
+import dev._2lstudios.chatsentinel.shared.chat.ChatPlayer;
+import dev._2lstudios.chatsentinel.shared.chat.ChatPlayerManager;
+import dev._2lstudios.chatsentinel.shared.interfaces.Module;
+import dev._2lstudios.chatsentinel.shared.modules.*;
+import dev._2lstudios.chatsentinel.shared.utils.StringUtil;
+import dev._2lstudios.chatsentinel.shared.utils.VersionUtil;
+import dev._2lstudios.chatsentinel.velocity.ChatSentinel;
+import dev._2lstudios.chatsentinel.velocity.modules.ModuleManager;
+import net.kyori.adventure.text.Component;
+
+import java.util.UUID;
+import java.util.regex.Pattern;
+
+public class ChatListener {
+	final private ProxyServer proxyServer;
+	final private ModuleManager moduleManager;
+	final private ChatPlayerManager chatPlayerManager;
+
+	public ChatListener(final ProxyServer proxyServer, final ModuleManager moduleManager,
+                        final ChatPlayerManager chatPlayerManager) {
+		this.proxyServer = proxyServer;
+		this.moduleManager = moduleManager;
+		this.chatPlayerManager = chatPlayerManager;
+	}
+
+	private void sendMessage(final CommandSource source, final String message) {
+		source.sendMessage(Component.text(message.replace('&', '§')));
```

Comment:
Here you can use adventure's LegacyComponentSerializer:
```java
LegacyComponentSerializer.legacyAmpersand()
```

Reply:
Updated in the last push, Thanks!